core/src/main/java/org/mule/runtime/core/DefaultMuleContext.java

```diff
@@ -254,22 +252,7 @@
   private ProcessorInterceptorProvider processorInterceptorManager;
 
   static {
-    // Ensure reactor operatorError hook is always registered.
-    Hooks.onOperatorError((throwable, signal) -> {
-      // Unwrap all throwables to be consistent with reactor default hook.
-      throwable = unwrap(throwable);
-      // Only apply hook for Event signals.
-      if (signal instanceof Event) {
-        return throwable instanceof MessagingException ? throwable
-            : new MessagingException((Event) signal, getRootCauseException(throwable));
-      } else {
-        return throwable;
-      }
-    });
-
-    // Log dropped events/errors rather than blow up which causes cryptic timeouts and stack traces.
-    Hooks.onErrorDropped(error -> logger.error("ERROR DROPPED UNEXPECTEDLY " + error));
-    Hooks.onNextDropped(event -> logger.error("EVENT DROPPED UNEXPECTEDLY " + event));
+    setupErrorHooks(Hooks.class);
```

Comment:
Can the container classlaoder do the initialization instead of having this call here?